core/src/test/java/greencity/controller/EcoNewsCommentControllerTest.java

```diff
@@ -0,0 +1,199 @@
+import greencity.service.EcoNewsCommentService;
+import greencity.service.UserService;
+import java.security.Principal;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableHandlerMethodArgumentResolver;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ContextConfiguration
+@Import(SecurityConfig.class)
+class EcoNewsCommentControllerTest {
+    private static final String ecoNewsCommentControllerLink = "/econews/comments";
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private EcoNewsCommentController ecoNewsCommentController;
+
+    @Mock
+    private EcoNewsCommentService ecoNewsCommentService;
+
+    @Mock
+    private UserService userService;
+
+    @BeforeEach
+    void setup() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(ecoNewsCommentController)
+            .setCustomArgumentResolvers(new PageableHandlerMethodArgumentResolver())
+            .build();
+    }
+
+    @Test
+    void save() throws Exception {
+        Principal principal = Mockito.mock(Principal.class);
+        when(principal.getName()).thenReturn("datskomar@gmail.com");
+        User user = getUser();
+        when(userService.findByEmail(anyString())).thenReturn(user);
+
+        String content = "{\n"
+            + "  \"parentCommentId\": 0,\n"
+            + "  \"text\": \"string\"\n"
+            + "}";
+
+        mockMvc.perform(post(ecoNewsCommentControllerLink + "/{econewsId}", 1)
+            .principal(principal)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(content))
+            .andExpect(status().isCreated());
+
+        ObjectMapper mapper = new ObjectMapper();
+        AddEcoNewsCommentDtoRequest addEcoNewsCommentDtoRequest =
+            mapper.readValue(content, AddEcoNewsCommentDtoRequest.class);
+
+        verify(userService).findByEmail(eq("datskomar@gmail.com"));
```

Comment:
Are you sure you want to put your real email in sources, especially in sources of an open source project. (Actually you already did it)